### PR TITLE
Article harmonisation

### PR DESCRIPTION
**Pull request by jinchad**
This is a pull request (PR), as instructed by @Wilsven under Trello card "[Create slides on GenAI development](https://trello.com/c/tAieMgFM/152-create-slides-on-genai-development)". With this PR, all team members can access the current prompts that I have created under `prompts.py`. Details of the files pertaining to this PR are listed below. Since this is the first PR, this description will only be describing what each file and folder in this project does. Subsequent PRs will contain the updates made to the files. 

## docs
- this folder contains all miscellaneous documents pertaining to this project
   ### images 
   - this is a sub-folder under `docs` contains images relevant to this project
## models.py
- this is a python file containing classes to instantiate each LLM used in the project.
## harmonisation.py
- this is a python file containing the graph and nodes.
- run this file to run the process of article harmonisation + optimisation
## prompt.py
- This file contains classes to instantiate prompts unique to each LLM type. 
- @jzkoh @jwyting @Wilsven kindly refer to this file for the prompts used in this project
## requirements.txt
- this is a txt file containing all the packages needed to run the project. Kindly run `pip install requirements.txt` before running any files in the project folder
## README.md
- this file will consist of some simple documentation of the project

### CODE REVIEW

**Comment by Wilsven** [comment]
Please check the GitHub Actions to debug the issue. I believe `ruff` does not check `.ipynb` notebooks.

**Comment by Wilsven** [comment]
> Please check the GitHub Actions to debug the issue. I believe `ruff` does not check `.ipynb` notebooks.

It seems like you have successfully debugged the issue. Feel free to ignore this comment.

File: .gitignore [hunk 1]
```diff
@@ -11,6 +11,7 @@ data/*
 # Exclude these personal folders and files
 experiments
 *.model
+.env
```
Comment (Wilsven) [inline .gitignore:14]: Did you check if the `.gitignore` already contains this? I'm pretty sure it does so please check and modify if necessary.

File: .gitignore [hunk 2]
```diff
@@ -171,6 +172,8 @@ venv/
 ENV/
 env.bak/
 venv.bak/
+hh_venv/
```
Comment (Wilsven) [inline .gitignore:175]: It's standard practice to name your virtual environment `venv` which would have already been taken care of by the `.gitignore`. So please modify this. Thanks.

File: article-harmonisation/requirements.txt
Comment (Wilsven) [inline article-harmonisation/requirements.txt:1]: Are you sure your project requires this many packages?

File: article-harmonisation/README.md
Comment (Wilsven) [inline article-harmonisation/README.md:1]: `README.md` seems incomplete. You have provided instructions on how to get the `HUGGINGFACEHUB_API_TOKEN` but what about the `LANGCHAIN_API_KEY`? Please also make sure your instructions in the `README.md` is as clear as possible.

**Review by Wilsven — CHANGES_REQUESTED** [review]
I have left some comments for review. Please resolve before making another PR. @jinchad

File: .gitignore (hunk 1 above)
Comment (jinchad) [reply .gitignore]: removed

File: .gitignore (hunk 2 above)
Comment (jinchad) [reply .gitignore]: deleted `hh_venv` and created a new venv called `venv`

File: article-harmonisation/requirements.txt
Comment (jinchad) [reply article-harmonisation/requirements.txt:1]: new requirements.txt will have lesser packages

File: article-harmonisation/README.md
Comment (jinchad) [reply article-harmonisation/README.md:1]: `LANGCHAIN_API_KEY` no longer needed, hence removed.

File: article-harmonisation/requirements.txt
Comment (Wilsven) [reply article-harmonisation/requirements.txt:1]: Care to elaborate?

Comment (jinchad) [reply article-harmonisation/requirements.txt:1]: Sure, i have created a new venv, and installed the packages one by one again while ensuring that the files can still run smoothly. I got a requirements.txt at the end with lesser packages.

**Comment by Wilsven** [comment]
@jinchad Please resolve the merge conflicts and have proper description of your PR. For now, I am unclear what this PR is about. From what I know about what you've commented in Trello, it seems you're trying to share the prompt templates you've used? It is currently unclear. Please fix and re-request review at the top right corner. Or please close this PR and make a new one. Thanks.

File: article-harmonisation/requirements.txt
Comment (Wilsven) [reply article-harmonisation/requirements.txt:1]: Thanks, I will resolve this conversation then.

**Comment by jinchad** [comment]
@Wilsven This PR was done as per your instructions under the Trello card "Create Slides on GenAI Development" which is under Project Management now.

**Comment by Wilsven** [comment]
> @Wilsven This PR was done as per your instructions under the Trello card "Create Slides on GenAI Development" which is under Project Management now.

image

Your description on your PR should go here. Please click into the tab "Files changed" and give proper description about each of the changes you're committing to this PR. It is currently unclear. 

**Note:** Do resolve the merge conflict of the `CHANGELOG.md` as well. Thanks. Please take note not to overwrite anyone's changes. Check before you confirm something permanent.

File: CHANGELOG.md
Comment (Wilsven) [inline CHANGELOG.md:1]: There's a merge conflict for this file. Please make the necessary amends.

File: pyvenv.cfg
Comment (Wilsven) [inline pyvenv.cfg:1]: Is this meant to be committed into Version Control?

**Review by Wilsven — CHANGES_REQUESTED** [review]

File: CHANGELOG.md
Comment (issacj-17) [reply CHANGELOG.md:1]: Yeap, I will do this. There was a bug when I was checking out my branch and merging some changes to kedro-text-preprocess.

**Comment by Wilsven** [comment]
> PR request to share prompts as instructed by @Wilsven on Trello card "Create Slides on GenAI Development" under Project Management

@jinchad Just to let you descriptions for your PR are meant to be informative. This is bad practice if you're going to type something arbitrary. But if you're going to do it nonetheless, please at least add the link to the Trello card. 

You may consult @issacj-17 on how he provides informative descriptions to his PRs. I believe he does it pretty well so it would be a good learning opportunity for you. Thanks all!

**Review by Wilsven — APPROVED** [review]

**Comment by Wilsven** [comment]
@jinchad @issacj-17 Before we merge this PR into `main`, is there a documentation somewhere on how these can be run? Or is that still in the works?

**Comment by jinchad** [comment]
@Wilsven I have updated the description in this PR and took advice from @issacj-17. Will have more informative descriptions in future PR from now on. To answer your question on the documentation on how to run the project: the project can be run after installing the packages in `requirements.txt` and running `harmonisation.py`

**Comment by Wilsven** [comment]
> @Wilsven I have updated the description in this PR and took advice from @issacj-17. Will have more informative descriptions in future PR from now on. To answer your question on the documentation on how to run the project: the project can be run after installing the packages in `requirements.txt` and running `harmonisation.py`. I will add that into the `README.md`

Nice job, it is very well done @jinchad . Also, thanks @issacj-17. 

Once the `README.md` has been updated on how to run the `harmonisation.py` file, I will merge the PR.

**Comment by jinchad** [comment]
@Wilsven The README has been updated with instructions on how to run the project. Thank you.

**Comment by Wilsven** [comment]
Give me some time. Will review and merge if there are no more changes to be made. Thank you.

**Pull request merged**